agera/src/main/java/com/google/android/agera/CompiledRepository.java

```diff
@@ -225,13 +231,20 @@ private boolean checkCancellationLocked() {
    */
   void acknowledgeCancel() {
     boolean shouldStartFlow = false;
+    Object discardedIntermediateValue = null;
     synchronized (this) {
       if (runState == CANCEL_REQUESTED) {
         runState = IDLE;
+        if (intermediateValue != currentValue && intermediateValue != initialValue) {
```

Comment:
any additional paths to cover in the unit tests on this branch (and same check in other methods)?